routingv8/routes.go

```diff
@@ -65,6 +66,29 @@ func (s *RoutingService) Routes(
 		}
 		values.Add("avoid[features]", strings.Join(areas, ","))
 	}
+	if req.Vehicle != nil {
+		if req.Vehicle.GrossWeight != 0 {
+			values.Add("vehicle[grossWeight]", strconv.Itoa(req.Vehicle.GrossWeight))
+		}
+		if req.Vehicle.TrailerCount != 0 {
+			values.Add("vehicle[trailerCount]", strconv.Itoa(req.Vehicle.TrailerCount))
+		}
+		if req.Vehicle.AxleCount != 0 {
+			values.Add("vehicle[axleCount]", strconv.Itoa(req.Vehicle.AxleCount))
+		}
+		if req.Vehicle.Height != 0 {
+			values.Add("vehicle[height]", strconv.Itoa(req.Vehicle.Height))
+		}
+		if req.Vehicle.Width != 0 {
+			values.Add("vehicle[width]", strconv.Itoa(req.Vehicle.Width))
+		}
+		if req.Vehicle.Length != 0 {
+			values.Add("vehicle[length]", strconv.Itoa(req.Vehicle.Length))
+		}
+		if req.Vehicle.Type != "" {
+			values.Add("vehicle[type]", req.Vehicle.Type.String())
+		}
+	}
```

Comment:
Isn't 0 and allowed value in many of these?

Reply:
I'm using the 0 value as a presence check here. I'm not sure what the difference between setting height to zero or not setting it at all, for example. But this felt easier than to fiddle with pointers for every field to check for presence. Or do you think it's better to always set everything?

Reply:
If we always set everything then we have to make sure to set sane defaults as some values can only be 1 or higher, for example. I am not sure if HERE handles presence differently than, for example, 0 but in many of these fields they explicitly document the default to be 0 so maybe this is good enough for now.